service/service.go

```diff
@@ -98,16 +99,43 @@ func New(tarball io.Reader, env map[string]string, options ...Option) (*Service,
 	if err := s.setOptions(options...); err != nil {
 		return nil, err
 	}
+	// untar tarball to retrieve mesg.yml
 	if err := s.saveContext(tarball); err != nil {
 		return nil, err
 	}
 	defer os.RemoveAll(s.tempPath)
-
 	def, err := importer.From(s.tempPath)
 	if err != nil {
 		return nil, err
 	}
 
+	// XXX: remove .git folder from repo.
+	// It makes docker build iamge id same between repo clones.
+	if err := os.RemoveAll(filepath.Join(s.tempPath, ".git")); err != nil {
+		return nil, err
+	}
+
+	// XXX: change the access and modification times of service to get constant hash
+	zerot := time.Time{}
+	if err := filepath.Walk(s.tempPath, func(path string, _ os.FileInfo, _ error) error {
```

Comment:
I wonder if this is really useful. Git and tar concern the timestamp of file, don't they?

Reply:
Yes, the do. WIthout it ever git clone will be diffrent :/

Reply:
Why is it different, we had this problem before and removing the .git was the solution because we had a file from the logs in git that was different. You can look this PR here https://github.com/mesg-foundation/core/pull/482.

Reply:
It's different problem. Git was writing timestamp to the file. Here tar save mod/access/change time of the files.

You can check out [Header](https://golang.org/pkg/archive/tar/#Header) for tar archive file